org.eclipse.scanning.connector.epics/src/org/eclipse/scanning/connector/epics/EpicsV4ConnectorService.java

```diff
@@ -336,17 +338,17 @@ private MalcolmMessage sendCallMessage(IMalcolmDevice<?> device, MalcolmMessage
 			throw new Exception(errMEssage);
 	        }
 
-			logger.debug("Call method = \n" + methodStructure + "\nEND");
+			logger.debug("Call method = \n{}\nEND", methodStructure);
 	        PvaClientRPC rpc = pvaChannel.createRPC(methodStructure);
 	        rpc.issueConnect();
 	        status = rpc.waitConnect();
 	        if(!status.isOK()) {
 			String errMEssage = "CreateRPC failed for '" + message.getMethod() + "' (" + status.getType() + ": " + status.getMessage() + ")";
 			throw new Exception(errMEssage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)